requirements.txt

```diff
@@ -15,6 +15,66 @@ ssh-python>=0.6.0
 parallel-ssh==2.2.0
 requests==2.22.0
 wheel>=0.32.0
-gevent
-greenlet
+gevent~=21.1.2
+greenlet~=1.0.0
 fasteners==0.16
+
+py~=1.10.0
+colorama~=0.4.3
+beautifulsoup4~=4.9.3
+soupsieve~=2.2.1
+pip~=20.2.1
+google~=3.0.0
+pluggy~=0.13.1
+six~=1.15.0
+toml~=0.10.2
+packaging~=21.0
+pytest~=6.2.4
+attrs~=21.2.0
+setuptools~=49.2.1
+cffi~=1.14.5
+behave~=1.2.6
+parse~=1.19.0
+geomet~=0.2.1.post1
+iniconfig~=1.1.1
+atomicwrites~=1.4.0
+certifi~=2020.12.5
+chardet~=3.0.4
+urllib3~=1.25.11
+paramiko~=2.7.2
+bcrypt~=3.2.0
+idna~=2.8
+pyparsing~=2.4.7
+pycparser~=2.20
+
+jmespath~=0.9.3
+docutils~=0.14
+configparser~=3.8.1
+keyring~=23.0.1
+rsa~=3.4.2
+pyasn1~=0.4.2
+tox~=3.23.0
+filelock~=3.0.12
+nose~=1.3.7
+okta~=0.0.4
+botocore~=1.20.61
+boto3~=1.17.61
+s3transfer~=0.4.2
+awscli~=1.19.61
+flake8~=3.7.9
+entrypoints~=0.3
+pyflakes~=2.1.1
+coverage~=5.5
+pexpect~=4.2.1
+jeepney~=0.6.0
+SecretStorage~=3.3.1
+Pygments~=2.2.0
+Pillow~=5.1.0
+roman~=2.0.0
+simplejson~=3.13.2
+Twisted~=17.9.0
+virtualenv~=20.4.3
+distlib~=0.3.1
+appdirs~=1.4.4
+fido2~=0.9.1
+zipp~=3.4.1
```

Comment:
Why did you add all these as requirements? It looks like some of them broke the debian package builds, and some of them should definitely be kept out of here such as `google` or `boto3` for example, since they're backend specific.
We already have `requirements-s3.txt`  or `requirements-gcs.txt` which contain these dependencies through higher level ones.

Reply:
Good catch.   Picked up those during some local testing.  I need to use the pip `freeze` option before local changes.